File: BackEnd/Src/Conecta.API/Conecta.API/Models/MateriaProfessorTurma.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Conecta.API.Models
{
    public class MateriaProfessorTurma
    {
        public int MateriaProfessorTurmaId { get; set; }
        public int TurmaId { get; set; }
        public Turma Turma { get; set; }
        public int MateriaId { get; set; }
        public Materia Materia { get; set; }
        public int ProfessorId { get; set; }
        public Professor Professor { get; set; }
        public ICollection<Nota> Notas { get; set; }
    }
}

File: FrontEnd/Conecta.App/src/app/Components/materia-professor-turma/materiaProfessorTurma.ts
import { Turma } from './../turmas/Turma';
import { Professor } from './../professor/Professor';
import { Materia } from '../materia/Materia';
export class materiaProfessorTurma{
  materiaProfessorTurmaId: number;
  turmaId: number;
  materiaId: number;
  professorId: number;
  professor: Professor;
  turma: Turma;
  materia: Materia;
}
